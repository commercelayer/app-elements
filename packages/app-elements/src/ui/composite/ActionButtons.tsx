import { Button, type ButtonVariant } from '#ui/atoms/Button'
import { withSkeletonTemplate } from '#ui/atoms/SkeletonTemplate'
import { useMemo, type MouseEventHandler } from 'react'

export interface ActionButtonsProps {
  actions: Array<{
    label: string
    onClick: MouseEventHandler<HTMLButtonElement>
    variant?: ButtonVariant
    disabled?: boolean
  }>
}

export const ActionButtons = withSkeletonTemplate<ActionButtonsProps>(
  ({ actions }): JSX.Element | null => {
    const isPrimary = (
      action: ActionButtonsProps['actions'][number]
    ): boolean => action.variant == null || action.variant === 'primary'

    const primaryActions = useMemo(
      () => actions.filter((action) => isPrimary(action)),
      [actions]
    )

    const secondaryActions = useMemo(
      () => actions.filter((action) => !isPrimary(action)),
      [actions]
    )

    if (actions.length === 0) {
      return null
    }

    return (
      <div
        data-testid='action-buttons'
        className='flex justify-end border-b border-gray-100 py-6'
      >
        {primaryActions.length === 1 && secondaryActions.length === 0 ? (
          <>
            {primaryActions.map(({ label, ...props }) => (
              <Button className='w-full' key={label} {...props}>
                {label}
              </Button>
            ))}
          </>
        ) : (
          <>
            <div className='basis-1/2 flex gap-3'>
              {secondaryActions.map(({ label, ...props }) => (
                <Button key={label} {...props}>
                  {label}
                </Button>
              ))}
            </div>
            <div className='basis-1/2 flex gap-3 justify-end'>
              {primaryActions.map(({ label, ...props }) => (
                <Button key={label} {...props}>
                  {label}
                </Button>
              ))}
            </div>
          </>
        )}
      </div>
    )
  }
)

ActionButtons.displayName = 'ActionButtons'
