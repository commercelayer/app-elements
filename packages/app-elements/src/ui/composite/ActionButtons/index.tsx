export { ActionButtons } from './ActionButtons'
export type { ActionButtonsProps } from './ActionButtons'
