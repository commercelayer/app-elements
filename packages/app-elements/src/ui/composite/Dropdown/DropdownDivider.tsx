interface Props extends React.HTMLAttributes<HTMLElement> {
  children?: React.ReactNode
}

export function DropdownDivider({ children, ...rest }: Props): JSX.Element {
  return (
    <div {...rest} className='h-px'>
      <hr className='border-gray-600' />
    </div>
  )
}

DropdownDivider.displayName = 'DropdownDivider'
