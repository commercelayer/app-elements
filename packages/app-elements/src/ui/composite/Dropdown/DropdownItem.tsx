import { Icon, type IconProps } from '#ui/atoms/Icon/Icon'
import { type FC } from 'react'

export interface DropdownItemProps extends React.HTMLAttributes<HTMLElement> {
  label: string
  icon?: IconProps['name'] | 'keep-space'
}

export const DropdownItem: FC<DropdownItemProps> = ({
  label,
  icon,
  ...rest
}) => {
  return (
    <button
      {...rest}
      className='w-full bg-black text-white py-2 pl-4 pr-8 text-sm font-semibold cursor-pointer flex items-center hover:bg-primary focus:bg-primary focus:!outline-none'
      aria-label={label}
    >
      {icon != null ? (
        <div className='mr-2'>
          {icon === 'keep-space' ? (
            <FakeIcon /> // keep the gap as if there was an icon
          ) : (
            <Icon name={icon} weight='bold' />
          )}
        </div>
      ) : null}

      <span
        className='text-ellipsis overflow-hidden whitespace-nowrap'
        title={label}
      >
        {label}
      </span>
    </button>
  )
}
DropdownItem.displayName = 'DropdownItem'

const FakeIcon: FC = () => {
  return <div className='w-[14px]' />
}
