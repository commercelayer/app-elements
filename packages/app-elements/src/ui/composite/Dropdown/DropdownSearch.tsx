import { Icon } from '#ui/atoms/Icon'
import cn from 'classnames'
import debounce from 'lodash-es/debounce'
import { forwardRef, type JSX, useCallback, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'

export interface DropdownSearchProps {
  /**
   * Callback triggered when the user types in the search input.
   * By default, this callback is debounced by 500ms.
   */
  onSearch: (hint: string) => void
  /**
   * Debounce time in ms for the onSearch callback. Set to 0 to disable debounce.
   * @default 500
   */
  debounceMs?: number
  /**
   * Placeholder text for the input element
   */
  placeholder?: string
  /**
   * Enable auto focus on the input element
   */
  autoFocus?: boolean
}

/**
 * This component renders an input as dropdown menu item with debounced `onSearch` callback.
 */
export const DropdownSearch = forwardRef<HTMLInputElement, DropdownSearchProps>(
  (
    { onSearch, debounceMs = 500, placeholder, autoFocus, ...rest },
    ref
  ): JSX.Element => {
    const [searchValue, setSearchValue] = useState('')
    const { t } = useTranslation()

    const debouncedOnSearch = useCallback(debounce(onSearch, debounceMs), [
      onSearch
    ])

    useEffect(
      function unmountDebounce() {
        return () => {
          debouncedOnSearch?.cancel()
        }
      },
      [debouncedOnSearch]
    )

    return (
      <div className='relative w-full' {...rest}>
        <Icon
          name='magnifyingGlass'
          weight='bold'
          className='absolute top-1/2 left-4 transform -translate-y-1/2 text-gray-400 pointer-events-none select-none text-sm '
        />
        <input
          className={cn(
            'pl-10 pr-8 py-2 bg-transparent min-w-max font-semibold text-sm placeholder:text-gray-400 !ring-0'
          )}
          placeholder={placeholder ?? t('common.search')}
          value={searchValue}
          onChange={({ currentTarget: { value } }) => {
            setSearchValue(value)
            debouncedOnSearch(value)
          }}
          ref={ref}
          autoFocus={autoFocus}
        />
      </div>
    )
  }
)

DropdownSearch.displayName = 'DropdownSearch'
