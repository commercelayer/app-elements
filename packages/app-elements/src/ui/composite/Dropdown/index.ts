export { Dropdown } from './Dropdown'
export { DropdownDivider } from './DropdownDivider'
export { DropdownItem } from './DropdownItem'
