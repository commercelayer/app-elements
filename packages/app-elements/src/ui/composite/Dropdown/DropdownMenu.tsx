import cn from 'classnames'
import { type FC } from 'react'
import { DropdownDivider } from './DropdownDivider'

export interface DropdownMenuProps extends React.HTMLAttributes<HTMLElement> {
  /** Menu content */
  children?: React.ReactNode
  /** Set to `none` to hide the top arrow */
  arrow?: 'none'
  /** Optional header for the dropdown menu */
  menuHeader?: string
  /**
   * Opening position of the dropdown menu
   * @default bottom-right
   */
  menuPosition?: 'bottom-left' | 'bottom-right' | 'top-left' | 'top-right'
}

export const DropdownMenu: FC<DropdownMenuProps> = ({
  children,
  arrow,
  menuHeader,
  menuPosition = 'bottom-right',
  ...rest
}) => {
  return (
    <div
      className={cn('flex', {
        'flex-col items-end': menuPosition === 'bottom-right',
        'flex-col items-start': menuPosition === 'bottom-left',
        'flex-col-reverse items-end': menuPosition === 'top-right',
        'flex-col-reverse items-start': menuPosition === 'top-left'
      })}
    >
      {arrow === 'none' ? null : (
        <Arrow menuPosition={menuPosition} centerForWidth={32} />
      )}
      <div
        {...rest}
        className='bg-black text-white rounded min-w-[150px] overflow-hidden py-1 sm:max-w-[250px]'
      >
        {menuHeader != null && (
          <>
            <div
              className='py-2 px-4 text-gray-400 text-xs font-semibold text-ellipsis overflow-hidden whitespace-nowrap'
              title={menuHeader}
            >
              {menuHeader}
            </div>
            <DropdownDivider />
          </>
        )}
        {children}
      </div>
    </div>
  )
}

DropdownMenu.displayName = 'DropdownMenu'

const Arrow: FC<{
  menuPosition: DropdownMenuProps['menuPosition']
  centerForWidth: number
}> = ({ menuPosition = 'bottom-right', centerForWidth }) => {
  const arrowHeight = 5
  const arrowWidth = 12
  const centeringOffset = centerForWidth / 2 - arrowWidth / 2

  const alignProp = menuPosition.includes('right') ? 'right' : 'left'
  const arrowDirection =
    menuPosition === 'bottom-right' || menuPosition === 'bottom-left'
      ? 'top'
      : 'bottom'
  const cssForPointingDirection =
    arrowDirection === 'top'
      ? {
          borderBottomWidth: arrowHeight,
          borderTopColor: 'transparent'
        }
      : {
          borderTopWidth: arrowHeight,
          borderBottomColor: 'transparent'
        }

  return (
    <span
      className='relative border-black border-l-transparent border-r-transparent'
      style={{
        // base styles
        borderLeftWidth: arrowWidth / 2,
        borderRightWidth: arrowWidth / 2,
        ...cssForPointingDirection,
        // keep the arrow centered on the dropdown button
        [alignProp]: centeringOffset
      }}
    />
  )
}
