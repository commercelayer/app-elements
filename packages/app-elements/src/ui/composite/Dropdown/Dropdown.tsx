import { useClickAway } from '#hooks/useClickAway'
import { useOnBlurFromContainer } from '#hooks/useOnBlurFromContainer'
import { Button } from '#ui/atoms/Button'
import { type DropdownMenuProps } from '#ui/composite/Dropdown/DropdownMenu'
import { isSpecificJsxTag, isSpecificReactComponent } from '#utils/children'
import { CaretDown, DotsThreeCircle } from '@phosphor-icons/react'
import cn from 'classnames'
import { Children, cloneElement, useMemo, useState } from 'react'
import { DropdownMenu } from './DropdownMenu'

export interface DropdownProps
  extends Pick<DropdownMenuProps, 'menuHeader' | 'menuPosition' | 'menuWidth'> {
  /** The trigger for the dropdown menu. Can be a JSX Element or simply a `string`. */
  dropdownLabel?: React.ReactNode
  /** List of links and actions. You can use a combination of `DropdownItem` and `DropdownDivider` components. */
  dropdownItems: React.ReactNode
  /** Additional class name for the dropdown container. */
  className?: string
}

/**
 * Dropdown is toggleable, contextual overlay for displaying lists of links and actions.
 *
 * There are 2 components for rendering the list you can pass to the `dropdownItems` prop:
 * - `DropdownItem`: The trigger that handles dropdown selection.
 * - `DropdownDivider`: A visual separator for dropdown items.
 */
export const Dropdown: React.FC<DropdownProps> = ({
  dropdownLabel = <DotsThreeCircle size={32} />,
  menuHeader,
  dropdownItems,
  menuPosition = 'bottom-right',
  menuWidth,
  className
}) => {
  const [isExpanded, setIsExpanded] = useState(false)

  function toggle(): void {
    setIsExpanded(!isExpanded)
  }

  function close(): void {
    setIsExpanded(false)
  }

  const clickAwayRef = useClickAway(close)

  const closeDropdownMenuIfButtonClicked = (
    e: React.MouseEvent<HTMLElement>
  ): void => {
    if ((e.target as any).nodeName === 'BUTTON') {
      close()
    }
  }

  const handleBlur = useOnBlurFromContainer(close)

  if (Children.count(dropdownItems) === 0) {
    return null
  }

  const dropdownButton = useMemo(() => {
    if (
      isSpecificReactComponent(dropdownLabel, [/^Button$/]) ||
      isSpecificJsxTag(dropdownLabel, ['button'])
    ) {
      return cloneElement(dropdownLabel, {
        'aria-haspopup': true,
        'aria-expanded': isExpanded,
        onClick: () => {
          toggle()
        }
      })
    }

    return (
      <Button
        variant='link'
        aria-haspopup
        aria-expanded={isExpanded}
        className={cn('m-0 p-0 align-top', {
          '!text-black': typeof dropdownLabel !== 'string',
          '!no-underline hover:!underline': typeof dropdownLabel === 'string'
        })}
        onClick={() => {
          toggle()
        }}
      >
        {dropdownLabel}
        {typeof dropdownLabel === 'string' ? (
          <CaretDown className='inline-block ml-1 -mt-0.5' weight='bold' />
        ) : null}
      </Button>
    )
  }, [dropdownLabel, isExpanded])

  return (
    <div
      ref={isExpanded ? clickAwayRef : undefined}
      onBlur={handleBlur}
      className={cn('relative', className)}
    >
      {dropdownButton}
      {isExpanded && (
        <div
          className={cn('absolute z-30', {
            'top-full mt-[5px] right-0': menuPosition === 'bottom-right',
            'top-full mt-[5px] left-0': menuPosition === 'bottom-left',
            'bottom-full mb-[5px] right-0': menuPosition === 'top-right',
            'bottom-full mb-[5px] left-0': menuPosition === 'top-left'
          })}
          onClick={closeDropdownMenuIfButtonClicked}
        >
          <DropdownMenu
            menuHeader={menuHeader}
            menuPosition={menuPosition}
            parentElementRef={clickAwayRef}
            menuWidth={menuWidth}
          >
            {dropdownItems}
          </DropdownMenu>
        </div>
      )}
    </div>
  )
}

Dropdown.displayName = 'Dropdown'
