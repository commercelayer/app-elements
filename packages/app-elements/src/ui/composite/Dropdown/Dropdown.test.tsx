import { Button } from '#ui/atoms/Button'
import { Dropdown, DropdownDivider, DropdownItem } from '#ui/composite/Dropdown'
import { fireEvent, render } from '@testing-library/react'

const items = (
  <>
    <DropdownItem onClick={() => {}} icon='creditCard' label='Payments' />
    <DropdownItem
      onClick={() => {}}
      icon='keep-space' // no icon, keep same gap
      label='Items'
    />

    <DropdownDivider />
    <DropdownItem onClick={() => {}} icon='xCircle' label='Delete' />
  </>
)

describe('Dropdown', () => {
  test('Should use the provided Button when `dropdownLabel` is set to Button', () => {
    const { container, getByText, getAllByRole } = render(
      <Dropdown
        dropdownLabel={<Button>Open dropdown</Button>}
        dropdownItems={items}
      />
    )

    expect(container).toMatchInlineSnapshot(`
      <div>
        <div
          class="relative"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="rounded whitespace-nowrap leading-5 inline-block text-center transition-opacity duration-500 px-6 py-3 text-base font-bold bg-black border border-black text-white hover:opacity-80"
          >
            Open dropdown
          </button>
        </div>
      </div>
    `)

    const button = getByText('Open dropdown')
    expect(button).not.toHaveClass('m-0 p-0 align-top')

    expect(getAllByRole('button').length).toEqual(1)
  })

  test('Should be rendering with default bottom-right position', () => {
    const { container, getByText } = render(
      <Dropdown dropdownLabel='open dropdown' dropdownItems={items} />
    )
    fireEvent.click(getByText('open dropdown'))
    expect(container).toMatchSnapshot()
  })

  test('Should be rendering bottom-left', () => {
    const { container, getByText } = render(
      <Dropdown
        dropdownLabel='open dropdown'
        menuPosition='bottom-left'
        dropdownItems={<div />}
      />
    )
    fireEvent.click(getByText('open dropdown'))
    expect(container).toMatchSnapshot()
  })

  test('Should be rendering top-left', () => {
    const { container, getByText } = render(
      <Dropdown
        dropdownLabel='open dropdown'
        menuPosition='top-left'
        dropdownItems={<div />}
      />
    )
    fireEvent.click(getByText('open dropdown'))
    expect(container).toMatchSnapshot()
  })

  test('Should be rendering top-right', () => {
    const { container, getByText } = render(
      <Dropdown
        dropdownLabel='open dropdown'
        menuPosition='top-right'
        dropdownItems={<div />}
      />
    )
    fireEvent.click(getByText('open dropdown'))
    expect(container).toMatchSnapshot()
  })
})
