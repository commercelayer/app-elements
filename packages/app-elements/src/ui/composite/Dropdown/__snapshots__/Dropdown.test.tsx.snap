// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dropdown > Should be rendering bottom-left 1`] = `
<div>
  <div
    class="relative"
  >
    <button
      aria-expanded="true"
      aria-haspopup="true"
      class="m-0 p-0 align-top rounded text-center font-bold focus:outline-none text-primary hover:opacity-80"
    >
      open dropdown
      <svg
        class="inline-block ml-1 -mt-0.5"
        fill="currentColor"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M216.49,104.49l-80,80a12,12,0,0,1-17,0l-80-80a12,12,0,0,1,17-17L128,159l71.51-71.52a12,12,0,0,1,17,17Z"
        />
      </svg>
    </button>
    <div
      class="absolute z-30 top-full mt-[5px] left-0"
    >
      <div
        class="flex flex-col items-start"
      >
        <span
          class="relative border-black border-l-transparent border-r-transparent"
          style="border-left-width: 6px; border-right-width: 6px; border-bottom-width: 5px; border-top-color: transparent; left: 10px;"
        />
        <div
          class="bg-black text-white rounded min-w-[150px] overflow-hidden py-1 sm:max-w-[250px]"
        >
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dropdown > Should be rendering top-left 1`] = `
<div>
  <div
    class="relative"
  >
    <button
      aria-expanded="true"
      aria-haspopup="true"
      class="m-0 p-0 align-top rounded text-center font-bold focus:outline-none text-primary hover:opacity-80"
    >
      open dropdown
      <svg
        class="inline-block ml-1 -mt-0.5"
        fill="currentColor"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M216.49,104.49l-80,80a12,12,0,0,1-17,0l-80-80a12,12,0,0,1,17-17L128,159l71.51-71.52a12,12,0,0,1,17,17Z"
        />
      </svg>
    </button>
    <div
      class="absolute z-30 bottom-full mb-[5px] left-0"
    >
      <div
        class="flex flex-col-reverse items-start"
      >
        <span
          class="relative border-black border-l-transparent border-r-transparent"
          style="border-left-width: 6px; border-right-width: 6px; border-top-width: 5px; border-bottom-color: transparent; left: 10px;"
        />
        <div
          class="bg-black text-white rounded min-w-[150px] overflow-hidden py-1 sm:max-w-[250px]"
        >
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dropdown > Should be rendering top-right 1`] = `
<div>
  <div
    class="relative"
  >
    <button
      aria-expanded="true"
      aria-haspopup="true"
      class="m-0 p-0 align-top rounded text-center font-bold focus:outline-none text-primary hover:opacity-80"
    >
      open dropdown
      <svg
        class="inline-block ml-1 -mt-0.5"
        fill="currentColor"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M216.49,104.49l-80,80a12,12,0,0,1-17,0l-80-80a12,12,0,0,1,17-17L128,159l71.51-71.52a12,12,0,0,1,17,17Z"
        />
      </svg>
    </button>
    <div
      class="absolute z-30 bottom-full mb-[5px] right-0"
    >
      <div
        class="flex flex-col-reverse items-end"
      >
        <span
          class="relative border-black border-l-transparent border-r-transparent"
          style="border-left-width: 6px; border-right-width: 6px; border-top-width: 5px; border-bottom-color: transparent; right: 10px;"
        />
        <div
          class="bg-black text-white rounded min-w-[150px] overflow-hidden py-1 sm:max-w-[250px]"
        >
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dropdown > Should be rendering with default bottom-right position 1`] = `
<div>
  <div
    class="relative"
  >
    <button
      aria-expanded="true"
      aria-haspopup="true"
      class="m-0 p-0 align-top rounded text-center font-bold focus:outline-none text-primary hover:opacity-80"
    >
      open dropdown
      <svg
        class="inline-block ml-1 -mt-0.5"
        fill="currentColor"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M216.49,104.49l-80,80a12,12,0,0,1-17,0l-80-80a12,12,0,0,1,17-17L128,159l71.51-71.52a12,12,0,0,1,17,17Z"
        />
      </svg>
    </button>
    <div
      class="absolute z-30 top-full mt-[5px] right-0"
    >
      <div
        class="flex flex-col items-end"
      >
        <span
          class="relative border-black border-l-transparent border-r-transparent"
          style="border-left-width: 6px; border-right-width: 6px; border-bottom-width: 5px; border-top-color: transparent; right: 10px;"
        />
        <div
          class="bg-black text-white rounded min-w-[150px] overflow-hidden py-1 sm:max-w-[250px]"
        >
          <button
            aria-label="Payments"
            class="w-full bg-black text-white py-2 pl-4 pr-8 text-sm font-semibold cursor-pointer flex items-center hover:bg-primary focus:bg-primary focus:!outline-none"
          >
            <div
              class="mr-2"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224,44H32A20,20,0,0,0,12,64V192a20,20,0,0,0,20,20H224a20,20,0,0,0,20-20V64A20,20,0,0,0,224,44Zm-4,24V88H36V68ZM36,188V112H220v76Zm172-24a12,12,0,0,1-12,12H164a12,12,0,0,1,0-24h32A12,12,0,0,1,208,164Zm-68,0a12,12,0,0,1-12,12H116a12,12,0,0,1,0-24h12A12,12,0,0,1,140,164Z"
                />
              </svg>
            </div>
            <span
              class="text-ellipsis overflow-hidden whitespace-nowrap"
              title="Payments"
            >
              Payments
            </span>
          </button>
          <button
            aria-label="Items"
            class="w-full bg-black text-white py-2 pl-4 pr-8 text-sm font-semibold cursor-pointer flex items-center hover:bg-primary focus:bg-primary focus:!outline-none"
          >
            <div
              class="mr-2"
            >
              <div
                class="w-[14px]"
              />
            </div>
            <span
              class="text-ellipsis overflow-hidden whitespace-nowrap"
              title="Items"
            >
              Items
            </span>
          </button>
          <div
            class="h-px my-1"
          >
            <hr
              class="border-gray-600"
            />
          </div>
          <button
            aria-label="Delete"
            class="w-full bg-black text-white py-2 pl-4 pr-8 text-sm font-semibold cursor-pointer flex items-center hover:bg-primary focus:bg-primary focus:!outline-none"
          >
            <div
              class="mr-2"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M168.49,104.49,145,128l23.52,23.51a12,12,0,0,1-17,17L128,145l-23.51,23.52a12,12,0,0,1-17-17L111,128,87.51,104.49a12,12,0,0,1,17-17L128,111l23.51-23.52a12,12,0,0,1,17,17ZM236,128A108,108,0,1,1,128,20,108.12,108.12,0,0,1,236,128Zm-24,0a84,84,0,1,0-84,84A84.09,84.09,0,0,0,212,128Z"
                />
              </svg>
            </div>
            <span
              class="text-ellipsis overflow-hidden whitespace-nowrap"
              title="Delete"
            >
              Delete
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
